server/recipeAPI/spoonacularAPI/SpoonacularRecipeAPI.ts

```diff
@@ -0,0 +1,312 @@
+import axios from 'axios';
+import * as dotenv from 'dotenv';
+import { URLSearchParams } from 'url';
+import IncorrectIDFormat from '../../exceptions/IncorrectIDFormat';
+import NoParameterFound from '../../exceptions/NoParameterFound';
+import ParameterIsNotAllowed from '../../exceptions/ParameterIsNotAllowed';
+import IFoodAPI from '../../foodAPI/IFoodAPI';
+import IFood from '../../serverAPI/model/food/IFood';
+import IInstruction from '../../serverAPI/model/instruction/IInstruction';
+dotenv.config();
+
+import IRecipe from "../../serverAPI/model/recipe/IRecipe";
+import IRecipeAPI from "../IRecipeAPI";
+
+export default class SpoonacularRecipeAPI implements IRecipeAPI {
+    protected foodAPI: IFoodAPI;
+
+    protected apiKey: string;
+    protected host: string;
+
+    protected recipeSearchParameters: Set<string>;
+    protected recipeGetParameters: Set<string>;
+
+    protected dishTypes: Set<string>;
+
+    protected headers: any;
+
+    constructor(foodAPI: IFoodAPI) {
+        this.foodAPI = foodAPI;
+
+        this.apiKey = process.env.SPOONACULAR_API_KEY;
+        this.host = process.env.SPOONACULAR_HOST;
+
+        this.dishTypes = new Set([
```

Comment:
Are these taken from spoonacular directly? If so, can you add a comment and a link to their docs saying so?